### PR TITLE
Rename flux functions to luminosity and fix direction binning edge cases

### PR DESCRIPTION
**Pull request by lukeshingles**
## Summary
This PR refactors emission line flux calculations to use more accurate terminology (luminosity instead of flux), fixes critical edge cases in direction binning for packets, and improves handling of coordinate systems in model data.

### Terminology Updates
- Renamed `get_line_fluxes_from_packets()` → `get_line_luminosities_from_packets()` with clarifying docstring that values are in [erg/s] without distance scaling
- Renamed `get_line_fluxes_from_pops()` → `get_line_luminosities_from_pops()`
- Renamed `make_flux_ratio_plot()` → `make_luminosity_ratio_plot()`
- Updated column naming from `flux_*` to `lum_*` in output data
- Renamed internal variables from `fluxdata` to `lumdata` for consistency

### Direction Binning Fixes
- Fixed phi bin overflow when `cosphi == -1` (acos reaches exactly 2π) by clamping to `nphibins - 1`
- Fixed phi bin overflow when `cosphi == 1` in monotonic ascending mode
- Added comprehensive test coverage for unequal phi/costheta bin counts
- Added test for edge case where direction lands exactly at phi upper boundary
- Updated packet direction binning to use configurable bin counts instead of hardcoded 10x10

### Model Data Improvements
- Added `min_abs_coordinate()` helper function to correctly handle cells straddling coordinate planes (where min/max cross zero)
- Fixed `pos_r_min` calculation for 3D models to account for cells that straddle coordinate planes
- Fixed `logrho` calculation to clamp at -99 (empty cell marker) instead of producing -inf for zero density
- Fixed kinetic energy calculation to avoid double-counting radial component in multi-dimensional models
- Fixed `vel_*_on_c` scaling to exclude `vel_*_kmps` columns (which are in km/s, not cm/s)
- Added `phi_mirrored` column naming in `get_cell_angle()` to clarify it's opposite sense from packet phi

### Code Refactoring
- Extracted transition data lookup outside inner loop in `get_line_luminosities_from_pops()` for efficiency
- Improved error handling with better comments distinguishing missing transitions from empty cells
- Renamed `get_rankbatch_parquetfile()` → `get_packets_rankbatch_parquetfile()` and `get_estimators_rankbatch_parquetfile()` for clarity
- Extracted `add_derived_estimator_columns()` as reusable function with proper null handling
- Fixed spectrum unit conversion to support "erg" as input unit in `convert_unit_to_angstroms()`

### Bug Fixes
- Fixed modelmeta initialization in packet processing when reading model data
- Fixed polarization file handling to track time blocks separately instead of modifying array
- Fixed data size reporting for virtual packets to sum all batch sizes instead of assuming uniform
- Fixed `nnelement_*` null handling to fill with zero only for those columns while preserving nulls elsewhere
- Fixed deposition data reading to properly handle timestep column

### Test Additions
- Added `test_directionbins_unequal_bincounts()` for phi/costheta bin count mismatch
- Added `test_directionbins_phibin_upper_edge()` for phi boundary overflow
- Added `test_pos_r_min_straddling_cells()` for coordinate plane straddling
- Added `test_min_abs_coordinate()` for coordinate minimum calculation
- Added `test_vel_on_c_excludes_kmps_columns()` for velocity scaling
- Added `test_add_derived_estimator_columns_*()` tests for estimator column derivation
- Added `test_xunit_conversion_roundtrip()` for spectrum unit conversion
- Added `test_average_direction_bins_unequal_bincounts()` for direction bin averaging

## Notable Implementation Details
- Direction bin packing formula: `dirbin = costhetabin * nphibins + phibin

https://claude.ai/code/session_01Nj64iauPNymsq6G2Ax25bo

### CODE REVIEW

File: artistools/inputmodel/inputmodel_misc.py [hunk 1]
```diff
@@ -638,12 +649,11 @@ def add_derived_cols_to_modeldata(
...
-    # all coord system also have a radial component calculated, so ignore this
+    # get total kinetic energy from orthogonal components. Every coordinate system also gets a radial component, which
+    # would double-count the orthogonal ones, so only use "r" for 1D models where it is the sole axis
+    orthogonal_axes = ["r"] if dimensions == 1 else [ax for ax in axes if ax != "r"]
     dfmodel = dfmodel.with_columns(
-        kinetic_en_erg=(
-            pl.sum_horizontal(pl.col(f"kinetic_en_erg_{ax}") for ax in axes if (ax != "r" or dimensions == 1))
-        )
+        kinetic_en_erg=(pl.sum_horizontal(pl.col(f"kinetic_en_erg_{ax}") for ax in orthogonal_axes))
     )
```
[the remark this answers is not in the document]
Comment (lukeshingles) [reply artistools/inputmodel/inputmodel_misc.py:657]: Not making this change — `pl.sum_horizontal` accepts a generator by design. Its signature is:

```
(*exprs: 'IntoExpr | Iterable[IntoExpr]', ignore_nulls: bool = True) -> Expr
```

`Iterable[IntoExpr]` as a single positional argument is explicitly supported, and polars flattens it via `parse_into_list_of_expressions`. Verified the two forms produce identical results:

```python
df = pl.DataFrame({"a_x": [1.0], "a_y": [2.0], "a_z": [4.0]})
gen = df.with_columns(tot=pl.sum_horizontal(pl.col(f"a_{ax}") for ax in "xyz"))
lst = df.with_columns(tot=pl.sum_horizontal(*[pl.col(f"a_{ax}") for ax in "xyz"]))
gen["tot"].to_list()  # [7.0]
lst["tot"].to_list()  # [7.0]
gen.equals(lst)       # True
```

The generator form is also pre-existing here — this diff only changed the filter condition, not the call style — and the same pattern is used elsewhere in the file with selectors (`pl.sum_horizontal(cs.starts_with("deposition_"))`).

---
_Generated by [Claude Code](https://claude.ai/code)_
